asv_bench/benchmarks/frame_methods.py

```diff
@@ -420,14 +420,33 @@ def setup(self):
         self.df = DataFrame({'a': np.random.randint(-1 << 8, 1 << 8, n),
                              'b': np.random.choice(t, n),
                              'c': np.random.choice(xs, n)})
-        self.df2 = DataFrame(np.random.randn(1000, 100).astype(str)).T
+        # df2 will not have any duplicates
+        self.df2 = DataFrame(np.random.randn(100, 1000).astype(str))
+
+        df3 = DataFrame(np.random.randint(0, 10, (2 ** 18, 5)),
+                        columns=list('ABCDE'))
+        df3.loc[:, 'F'] = Series('', index=df3.index).str.cat(df3.astype(str))
+        self.df3 = df3
 
     def time_frame_duplicated(self):
         self.df.duplicated()
 
     def time_frame_duplicated_wide(self):
         self.df2.duplicated()
 
+    def time_frame_duplicated_wide_inverse(self):
+        # will take fastpath for no duplicates
+        self.df2.duplicated(return_inverse=True)
+
+    def time_frame_duplicated_mixed(self):
```

Comment:
Instead of the 3 separate benchmarks below, you can parameterize over the `return_inverse` and `keep` keyword arguments. 

Here's an example: https://github.com/h-vetinari/pandas/blob/094139bffa07c101882bbfbf35a1621eadc00adc/asv_bench/benchmarks/frame_methods.py#L219-L231

Reply:
Thanks for the hint, coming in the commit for the next feedback round.

Reply:
Is it normal that the times are then not split up depending on the parameter?

I get:
```
[...]
[ 16.67%] ··· Running frame_methods.Duplicated.time_frame_duplicated         469ms;...
[ 33.33%] ··· Running frame_methods.Duplicated.time_frame_duplicated_mixed   125ms;...
[ 50.00%] ··· Running frame_methods.Duplicated.time_frame_duplicated_wide    203ms;...
[...]
```

Reply:
I believe the semicolons are splitting the time depending on the parameter. The format usually looks better with `asv continuous`

Reply:
@mroeschke The output I posted _is_ an extract from `asv continuous`, but I hadn't looked into `asv compare yet`, where the results were hiding. :)

Reply:
However, I get no outputs anymore for `master` (tried again three times...) -- could it be that a single parameter failing (like `return_inverse=True` will do on master) fails the whole test?